### ISSUE TITLE
rust-analyzer fails to start? 

### ISSUE THREAD

**Issue by zbelial** [emacs-lsp/lsp-mode#1373]
If I choose to use rust-analyzer, lsp fails to start and prints 
```
Error: Error("invalid type: null, expected a boolean", line: 0, column: 0)
```

Does you have this problem too? 

I use lsp-mode 6.2.1 and the (almost) latest rust-analyer.

**Comment by brotzeit** [comment]
Do you have any customizations ?

**Comment by brotzeit** [comment]
Hmmm, I can't reproduce this. Can you please try it again only with lsp-mode and rust-mode(or rustic) in your init.el.

**Comment by zbelial** [comment]
Hi, thanks for your effort firstly. @brotzeit 
I tried to update lsp-mode (from 6.2.1 to 20200203.1719) and surprisingly it works now. :)

**Comment by brotzeit** [comment]
@yyoncho Is it possible that an outdated session file is the reason for this ?

**Comment by yyoncho** [comment]
> @yyoncho Is it possible that an outdated session file is the reason for this ?

No, because the error comes from the server. IMO it is wrong/out of date property which we used to send as part of initializationOptions or workspace/configuration which you have fixed since 6.2.1 release.

**Comment by yyoncho** [comment]
> initializationOptions or workspace/configuration

Or client capabilities - we have such issues with CCLS server as well in the past - the strongly typed servers like those written in rust/C++ are less forgiving if we mess up the schema and send null instead of true/false.
